### ISSUE TITLE
fix conda-forge install instructions

### ISSUE THREAD

**Issue by espenhgn** [LFPy/LFPy#192]
`conda config --add channels conda-forge`
`conda install lfpy`
should suffice

**Comment by paolobettotti** [comment]
I'm trying to install LFPy on UBUNTU18.04 LTS (on a VirtualBox Machine) but none of the methods listed on the docs https://lfpy.readthedocs.io/en/latest/readme.html#installation work.

Initially I tried with the conda package (after adding the conda-forge channel) but "conda install LFPy" stucks forever on "solving environment" 

the I tried pip's versions and also by cloning the git repo. All requirements seems satisfied:
> pip install -r requirements.txt --user
Requirement already satisfied: setuptools>=23.1.0 in /home/paolo/anaconda3/lib/python3.8/site-packages (from -r requirements.txt (line 2)) (50.3.1.post20201107)
Requirement already satisfied: numpy>=1.8 in /home/paolo/anaconda3/lib/python3.8/site-packages (from -r requirements.txt (line 3)) (1.19.2)
Requirement already satisfied: scipy>=0.14 in /home/paolo/anaconda3/lib/python3.8/site-packages (from -r requirements.txt (line 4)) (1.5.2)
Requirement already satisfied: Cython>=0.20 in /home/paolo/anaconda3/lib/python3.8/site-packages (from -r requirements.txt (line 5)) (0.29.21)
Requirement already satisfied: h5py>=2.5 in /home/paolo/anaconda3/lib/python3.8/site-packages (from -r requirements.txt (line 6)) (2.10.0)
Requirement already satisfied: mpi4py>=1.2 in /home/paolo/anaconda3/lib/python3.8/site-packages (from -r requirements.txt (line 7)) (3.0.3)
Requirement already satisfied: matplotlib>=2.0 in /home/paolo/anaconda3/lib/python3.8/site-packages (from -r requirements.txt (line 8)) (3.3.2)
Requirement already satisfied: MEAutility>=1.4.2 in /home/paolo/anaconda3/lib/python3.8/site-packages (from -r requirements.txt (line 9)) (1.4.8)
Requirement already satisfied: neuron>=7.6.4 in /home/paolo/anaconda3/lib/python3.8/site-packages (from -r requirements.txt (line 10)) (7.8.1.1)
Requirement already satisfied: LFPykit>=0.2 in /home/paolo/anaconda3/lib/python3.8/site-packages (from -r requirements.txt (line 11)) (0.2)
Requirement already satisfied: six in /home/paolo/anaconda3/lib/python3.8/site-packages (from h5py>=2.5->-r requirements.txt (line 6)) (1.15.0)
Requirement already satisfied: pillow>=6.2.0 in /home/paolo/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0->-r requirements.txt (line 8)) (8.0.1)
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in /home/paolo/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0->-r requirements.txt (line 8)) (2.4.7)
Requirement already satisfied: cycler>=0.10 in /home/paolo/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0->-r requirements.txt (line 8)) (0.10.0)
Requirement already satisfied: python-dateutil>=2.1 in /home/paolo/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0->-r requirements.txt (line 8)) (2.8.1)
Requirement already satisfied: certifi>=2020.06.20 in /home/paolo/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0->-r requirements.txt (line 8)) (2020.6.20)
Requirement already satisfied: kiwisolver>=1.0.1 in /home/paolo/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0->-r requirements.txt (line 8)) (1.3.0)
Requirement already satisfied: pyyaml in /home/paolo/anaconda3/lib/python3.8/site-packages (from MEAutility>=1.4.2->-r requirements.txt (line 9)) (5.3.1)

but then I got:
> python setup.py develop
Creating x86_64 directory for .o files.

/home/paolo/GIT/LFPy/LFPy/test
expsyni.mod sinsyn.mod
expsyni.mod sinsyn.mod
/home/paolo/anaconda3/lib/python3.8/site-packages/neuron/.data/bin/nrnivmodl: 175: make: not found
Traceback (most recent call last):
  File "setup.py", line 48, in <module>
    spawn([find_executable('nrnivmodl')])
  File "/home/paolo/anaconda3/lib/python3.8/distutils/spawn.py", line 36, in spawn
    _spawn_posix(cmd, search_path, dry_run=dry_run)
  File "/home/paolo/anaconda3/lib/python3.8/distutils/spawn.py", line 157, in _spawn_posix
    raise DistutilsExecError(
distutils.errors.DistutilsExecError: command '/home/paolo/anaconda3/bin/nrnivmodl' failed with exit status 127

**Comment by espenhgn** [comment]
Hi, can you issue `conda install make` and try again? `nrnivmodl` looks for it.

**Comment by espenhgn** [comment]
As for 
> "conda install LFPy" stucks forever on "solving environment"

that could be explained by the complex task of trying to match versions of different packages if your environment has many different packages already. However, if your conda environment is newly created, I will have to look into it.

**Comment by paolobettotti** [comment]
> Hi, can you issue `conda install make` and try again? `nrnivmodl` looks for it.

no way. same behavior with a lot of conflicts.

**Comment by espenhgn** [comment]
That's odd. I think you have to document the exact steps taken for me to reproduce the error as well as the error message(s) and post those here. Also post the full output of `conda info` and `conda list` in the conda environment you've created. I recommend creating a separate environment for your project that require LFPy etc. to avoid this problem of "unsolvable environment". 
 
On a VM with Ubuntu 18.04 and miniconda 4.9.2 (just a lightweight version of anaconda), the following steps works without error on my end:
```
$ conda config --add channels conda-forge
$ conda config --set channel_priority strict
$ conda create -n "lfpy" python=3.8 pip wheel lfpy
$ conda activate lfpy
$ python -c "import LFPy; print(LFPy.__version__)"
2.1.2
$ python -c "import LFPy; LFPy.run_tests()"
mechanisms already loaded ...
<bunch of test output, some warnings but no errors>
```
That version of LFPy is identical to the latest stable release from PiPY and all dependencies are supposed to be installed as well (NEURON, MPI, h5py etc.). 

If you insist on using pip in a conda environment, the following procedure would install LFPy and most Python dependencies - but Python, compilers and some binaries are obtained from conda (mpi4py and h5py can be tedious to pip install):
```
$ conda deactivate
$ conda create -n "lfpypip" python=3.8 c-compiler cxx-compiler mpi4py h5py pip wheel
$ conda activate lfpypip
$ pip install lfpy
$ python -c "import LFPy; print(LFPy.__version__)"
2.1.2
...
```

**Comment by paolobettotti** [comment]
I reinstall all from scratch and your instructions (first set) worked perfectly. Many thanks!

**Comment by espenhgn** [comment]
> I reinstall all from scratch and your instructions (first set) worked perfectly. Many thanks!

Great! Good to hear it worked out.

**Comment by espenhgn** [comment]
Closing, merged #282 which should make things a bit more clear with regards to conda envs.
